tests/neuro_san/client/data_driven_agent_test.py

```diff
@@ -140,20 +141,49 @@ def interact(self, agent: str, session: AgentSession, interaction: Dict[str, Any
 
         # Evaluate response
         response: Dict[str, Any] = interaction.get("response", empty)
-
-        # Look for a block for each ChatMessage section key to test
-        for test_key in self.TEST_KEYS:
-            response_section: Dict[str, Any] = response.get(test_key, empty)
-            for one_evaluation, verify_for in response_section.items():
-                # Evaluate each item in the test block
-                evaluator: AgentEvaluator = AgentEvaluatorFactory.create_evaluator(self.asserts,
-                                                                                   one_evaluation,
-                                                                                   test_key)
-                evaluator.evaluate(processor, verify_for)
+        response_extractor = DictionaryExtractor(response)
+        self.test_response_keys(processor, response_extractor, self.TEST_KEYS)
 
         # See how we should continue the conversation
         return_chat_context: Dict[str, Any] = None
         if interaction.get("continue_conversation", True):
             return_chat_context = processor.get_chat_context()
 
         return return_chat_context
+
+    def test_response_keys(self, processor: BasicMessageProcessor,
+                           response_extractor: DictionaryExtractor,
+                           keys: List[str]):
+        """
+        Tests the given response keys
+
+        :param processor: The BasicMessageProcessor instance to query results from.
+        :param response_extractor: The DictionaryExtractor for the test structure from the test hocon file.
+        :param keys: The response keys to test
+        """
+        deeper_test_keys: List[str] = []
+
+        for test_key in keys:
+
+            test_key_value: Dict[str, Any] = response_extractor.get(test_key)
+            if test_key_value is None:
+                # Got nothing for test_key. Nothing to see here. Please move along.
+                continue
+
+            if isinstance(test_key_value, Dict):
+                # The value refers to a deeper dictionary test
+                for deeper_key in test_key_value.keys():
+                    deeper_test_keys.append(f"{test_key}.{deeper_key}")
```

Comment:
Accumulate fully specified keys like:
"sly_data.foo_key"  and "sly_data.foo_dict.foo_key" so these can all be tested with the same evaluators.